daml-lf/transaction/src/main/protobuf/com/daml/lf/transaction.proto

```diff
@@ -127,3 +127,19 @@ message NodeRollback { // *since version 14*
 }
 
 // architecture-handbook-entry-end: Nodes
+
+message Versioned {
+  string version = 1;
+  bytes payload = 2;
+}
+
+message FatContractInstance {
+  bytes contract_id = 1;
+  com.daml.lf.value.Identifier template_id = 2;
+  bytes create_arg = 3;
+  KeyWithMaintainers contract_key_with_maintainers = 4;
+  repeated string non_maintainer_signatories = 5;
+  repeated string non_signatory_stakeholders = 6;
```

Comment:
Add comments wrt the invariants required from a valid `FatContractInstance`: sorted and unique

Reply:
added in the transaction specification